addons/project/controllers/portal.py

```diff
@@ -70,13 +81,28 @@ def portal_my_projects(self, page=1, date_begin=None, date_end=None, sortby=None
         })
         return request.render("project.portal_my_projects", values)
 
-    @http.route(['/my/project/<int:project_id>'], type='http', auth="user", website=True)
-    def portal_my_project(self, project_id=None, **kw):
-        project = request.env['project.project'].browse(project_id)
-        vals = {'project': project}
-        history = request.session.get('my_projects_history', [])
-        vals.update(get_records_pager(history, project))
-        return request.render("project.portal_my_project", vals)
+    @http.route(['/my/project/<int:project_id>'], type='http', auth="public", website=True)
+    def portal_my_project(self, project_id=None, access_token=None, **kw):
+        try:
+            project_sudo = self._document_check_access('project.project', project_id, access_token)
+        except AccessError:
+            return request.redirect('/my')
+
+        values = self._project_get_page_view_values(project_sudo, access_token, **kw)
+        return request.render("project.portal_my_project", values)
+
+
+
+    # ------------------------------------------------------------
+    # My Task
+    # ------------------------------------------------------------
+    def _task_get_page_view_values(self, task, access_token, **kwargs):
+        values = {
+            'page_name': 'task',
+            'task': task,
+            'user': request.env.user
```

Comment:
Isn't it already in the rendered values ? I think qweb automaticallly adds it. To check.

Reply:
Cannot make it work without giving the user to the template.